src/http/input/RequestParser.ts

```diff
@@ -1,8 +1,18 @@
-import type { HttpRequest } from '../../server/HttpRequest';
-import { AsyncHandler } from '../../util/handlers/AsyncHandler';
-import type { Operation } from '../Operation';
+import type { BodyParser } from './body/BodyParser';
+import type { ConditionsParser } from './conditions/ConditionsParser';
+import type { TargetExtractor } from './identifier/TargetExtractor';
+import type { MetadataParser } from './metadata/MetadataParser';
+import type { PreferenceParser } from './preferences/PreferenceParser';
 
 /**
- * Converts an incoming HttpRequest to an Operation.
+ * Stores the classes necessary to create an Operation.
  */
-export abstract class RequestParser extends AsyncHandler<HttpRequest, Operation> {}
+export class RequestParser {
```

Comment:
So why is this not abstract? Weird.

Reply:
In this version, this class is pretty much the same as an interface, but is done like this so we can create it separately in Components.js. If it was abstract I think we would get errors there.